services/chatapi/api.go

```diff
@@ -0,0 +1,152 @@
+package chatapi
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/status-im/status-console-client/protocol/client"
+	"github.com/status-im/status-console-client/protocol/v1"
+)
+
+type PrivateAPI struct {
+	s *Service
+}
+
+func NewPrivateAPI(s *Service) *PrivateAPI {
+	return &PrivateAPI{
+		s: s,
+	}
+}
+
+func (api *PrivateAPI) Chats() (ChatsResponse, error) {
+	var result ChatsResponse
+
+	contacts, err := api.s.messenger.Contacts()
+	if err != nil {
+		return result, fmt.Errorf("failed to load contacts: %v", err)
+	}
+
+	contactToChat := make(map[client.Contact][]*protocol.Message)
+
+	for _, c := range contacts {
+		messages, err := api.s.messenger.Messages(c, 0)
```

Comment:
we will load whole db into memory. i think we need specific api to count unread messages per chat and load last msg per chat

Reply:
Yes, I will change that later.